src/Controller/Component/CsrfComponent.php

```diff
@@ -56,7 +56,7 @@ class CsrfComponent extends Component
         'cookieName' => 'csrfToken',
         'expiry' => 0,
         'secure' => false,
-        'httpOnly' => false,
+        'httpOnly' => true,
```

Comment:
@lorenzo  If the attacker submits a form, then the browser will forward the user's session cookie. I agree that an XSS exploit will let the attacker get the token out of the form. 

By having the cookie be HttpOnly they cannot modify or overwrite the cookie we set. When implementing the CSRF approach I tried to follow [the double submit cookies approach](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_%28CSRF%29_Prevention_Cheat_Sheet#Double_Submit_Cookies) which implies that the cookie should not be writable from the client, but missed the httponly part of the cookie.